// @angular/material/tooltip/typings/index.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from '@angular/material/tooltip';
import * as i2 from '@angular/common';
import * as i3 from '@angular/cdk/platform';
import * as i4 from '@angular/cdk/scrolling';
import * as i5 from '@angular/cdk/overlay';
import * as i6 from '@angular/cdk/bidi';
import * as i7 from '@angular/platform-browser';
import * as i8 from '@angular/cdk/a11y';
import * as i9 from '@angular/cdk/portal';
import * as i10 from '@angular/material/core';
export var MatTooltipModuleNgFactory = i0.ɵcmf(i1.MatTooltipModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, [TooltipComponentNgFactory]], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]),
        i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(4608, i3.Platform, i3.Platform, []), i0.ɵmpd(5120, i4.ScrollDispatcher, i4.SCROLL_DISPATCHER_PROVIDER_FACTORY, [[3, i4.ScrollDispatcher], i0.NgZone,
            i3.Platform]), i0.ɵmpd(5120, i4.ViewportRuler, i4.VIEWPORT_RULER_PROVIDER_FACTORY, [[3, i4.ViewportRuler], i3.Platform, i0.NgZone, i4.ScrollDispatcher]), i0.ɵmpd(4608, i5.ScrollStrategyOptions, i5.ScrollStrategyOptions, [i4.ScrollDispatcher,
            i4.ViewportRuler]), i0.ɵmpd(5120, i5.OverlayContainer, i5.ɵa, [[3, i5.OverlayContainer]]),
        i0.ɵmpd(4608, i5.ɵf, i5.ɵf, [i4.ViewportRuler]), i0.ɵmpd(4608, i5.Overlay, i5.Overlay, [i5.ScrollStrategyOptions, i5.OverlayContainer, i0.ComponentFactoryResolver,
            i5.ɵf, i0.ApplicationRef, i0.Injector, i0.NgZone]), i0.ɵmpd(5120, i5.ɵc, i5.ɵd, [i5.Overlay]), i0.ɵmpd(6144, i6.DIR_DOCUMENT, null, [i7.DOCUMENT]),
        i0.ɵmpd(4608, i6.Directionality, i6.Directionality, [[2, i6.DIR_DOCUMENT]]),
        i0.ɵmpd(4608, i8.InteractivityChecker, i8.InteractivityChecker, [i3.Platform]),
        i0.ɵmpd(4608, i8.FocusTrapFactory, i8.FocusTrapFactory, [i8.InteractivityChecker,
            i3.Platform, i0.NgZone]), i0.ɵmpd(136192, i8.AriaDescriber, i8.ARIA_DESCRIBER_PROVIDER_FACTORY, [[3, i8.AriaDescriber], i3.Platform]), i0.ɵmpd(5120, i8.LiveAnnouncer, i8.LIVE_ANNOUNCER_PROVIDER_FACTORY, [[3, i8.LiveAnnouncer], [2, i8.LIVE_ANNOUNCER_ELEMENT_TOKEN], i3.Platform]),
        i0.ɵmpd(5120, i8.FocusMonitor, i8.FOCUS_MONITOR_PROVIDER_FACTORY, [[3, i8.FocusMonitor],
            i0.NgZone, i3.Platform]), i0.ɵmpd(5120, i1.MAT_TOOLTIP_SCROLL_STRATEGY, i1.MAT_TOOLTIP_SCROLL_STRATEGY_PROVIDER_FACTORY, [i5.Overlay]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i9.PortalModule, i9.PortalModule, []), i0.ɵmpd(512, i3.PlatformModule, i3.PlatformModule, []), i0.ɵmpd(512, i4.ScrollDispatchModule, i4.ScrollDispatchModule, []), i0.ɵmpd(512, i5.OverlayModule, i5.OverlayModule, []),
        i0.ɵmpd(512, i10.CompatibilityModule, i10.CompatibilityModule, []),
        i0.ɵmpd(512, i6.BidiModule, i6.BidiModule, []), i0.ɵmpd(256, i10.MATERIAL_SANITY_CHECKS, true, []), i0.ɵmpd(512, i10.MatCommonModule, i10.MatCommonModule, [[2, i10.MATERIAL_SANITY_CHECKS]]), i0.ɵmpd(512, i8.A11yModule, i8.A11yModule, []), i0.ɵmpd(512, i1.MatTooltipModule, i1.MatTooltipModule, [])]);
});
var styles_TooltipComponent = ['.mat-tooltip-panel{pointer-events:none!important}.mat-tooltip{color:#fff;border-radius:2px;margin:14px;max-width:250px;padding-left:8px;padding-right:8px}@media screen and (-ms-high-contrast:active){.mat-tooltip{outline:solid 1px}}'];
export var RenderType_TooltipComponent = i0.ɵcrt({ encapsulation: 2,
    styles: styles_TooltipComponent, data: { 'animation': [{ type: 7, name: 'state', definitions: [{ type: 0,
                        name: 'initial, void, hidden', styles: { type: 6, styles: { transform: 'scale(0)' },
                            offset: null }, options: undefined }, { type: 0, name: 'visible',
                        styles: { type: 6, styles: { transform: 'scale(1)' }, offset: null }, options: undefined },
                    { type: 1, expr: '* => visible', animation: { type: 4, styles: null, timings: '150ms cubic-bezier(0.0, 0.0, 0.2, 1)' },
                        options: null }, { type: 1, expr: '* => hidden', animation: { type: 4, styles: null,
                            timings: '150ms cubic-bezier(0.4, 0.0, 1, 1)' }, options: null }],
                options: {} }] } });
export function View_TooltipComponent_0(_l) {
    return i0.ɵvid(2, [(_l()(), i0.ɵeld(0, null, null, 2, 'div', [['class',
                'mat-tooltip']], [[4, 'transform-origin', null], [24, '@state', 0]], [[null,
                '@state.start'], [null, '@state.done']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('@state.start' === en)) {
                var pd_0 = (_co._animationStart() !== false);
                ad = (pd_0 && ad);
            }
            if (('@state.done' === en)) {
                var pd_1 = (_co._animationDone($event) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), (_l()(), i0.ɵted(null, ['', '']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_2 = 'mat-tooltip';
        var currVal_3 = _co.tooltipClass;
        _ck(_v, 1, 0, currVal_2, currVal_3);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co._transformOrigin;
        var currVal_1 = _co._visibility;
        _ck(_v, 0, 0, currVal_0, currVal_1);
        var currVal_4 = _co.message;
        _ck(_v, 2, 0, currVal_4);
    });
}
export function View_TooltipComponent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'mat-tooltip-component', [['aria-hidden', 'true']], [[4, 'zoom', null]], [['body', 'click']], function (_v, en, $event) {
            var ad = true;
            if (('body:click' === en)) {
                var pd_0 = (i0.ɵnov(_v, 1)._handleBodyInteraction() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, View_TooltipComponent_0, RenderType_TooltipComponent)), i0.ɵdid(49152, null, 0, i1.TooltipComponent, [i0.ChangeDetectorRef], null, null)], null, function (_ck, _v) {
        var currVal_0 = ((i0.ɵnov(_v, 1)._visibility === 'visible') ? 1 : null);
        _ck(_v, 0, 0, currVal_0);
    });
}
export var TooltipComponentNgFactory = i0.ɵccf('mat-tooltip-component', i1.TooltipComponent, View_TooltipComponent_Host_0, {}, {}, []);
